include/multigrid_pclomp/multigrid_ndt_omp.h

```diff
@@ -0,0 +1,660 @@
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2010-2012, Willow Garage, Inc.
+ *  Copyright (c) 2012-, Open Perception, Inc.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id$
+ *
+ */
+
+#ifndef PCL_REGISTRATION_NDT_OMP_MULTI_VOXEL_H_
+#define PCL_REGISTRATION_NDT_OMP_MULTI_VOXEL_H_
+
+#include "boost/optional.hpp"
+
+#include <pcl/search/impl/search.hpp>
+#include <pcl/registration/registration.h>
+#include "multigrid_pclomp/multi_voxel_grid_covariance_omp.h"
+
+#include <unsupported/Eigen/NonLinearOptimization>
+
+namespace pclomp
+{
+	struct NdtResult
+	{
+		Eigen::Matrix4f pose;
+		float transform_probability;
+		float nearest_voxel_transformation_likelihood;
+		int iteration_num;
+		std::vector<Eigen::Matrix4f, Eigen::aligned_allocator<Eigen::Matrix4f>> transformation_array;
+		EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+	};
+
+	struct NdtParams
+	{
+		double trans_epsilon;
+		double step_size;
+		double resolution;
+		int max_iterations;
+		// pclomp::NeighborSearchMethod search_method;
+		int num_threads;
+		float regularization_scale_factor;
+	};
+
+	/** \brief A 3D Normal Distribution Transform registration implementation for point cloud data.
+	  * \note For more information please see
+	  * <b>Magnusson, M. (2009). The Three-Dimensional Normal-Distributions Transform —
+	  * an Efficient Representation for pcl::Registration, Surface Analysis, and Loop Detection.
```

Comment:
Probably an unintended change.
```suggestion
	  * an Efficient Representation for Registration, Surface Analysis, and Loop Detection.
```

Reply:
Resolved: https://github.com/tier4/ndt_omp/pull/19/commits/ca5e438585d50f27e635bbc343aaddbcf8288c41